packages/fetch-engine/src/download.ts

```diff
@@ -70,8 +70,12 @@ export async function download(options: DownloadOptions): Promise<BinaryPaths> {
   const platform = await getPlatform()
   const os = await getos()
 
-  if (os.targetDistro && ['nixos'].includes(os.targetDistro)) {
-    console.error(`${yellow('Warning')} Precompiled engine files are not available for ${os.targetDistro}.`)
+  if (os.targetDistro && ['nixos'].includes(os.targetDistro) && !allEngineEnvVarsSet(Object.keys(options.binaries))) {
+    console.error(
+      `${yellow('Warning')} Precompiled engine files are not available for ${
+        os.targetDistro
+      }, please provide the paths via environment variables, see https://www.prisma.io/docs/concepts/components/prisma-engines#using-custom-engine-libraries-or-binaries.`,
```

Comment:
Providing the link to the docs here is a great idea, thank you for doing this! Let's add a level of indirection via a shortlink to ensure the URL always stays relevant, and to make the console output cleaner, I'll open a PR in https://github.com/prisma/pris.ly

Reply:
```suggestion
      }, please provide the paths via environment variables, see https://pris.ly/d/custom-engines`,
```

Reply:
pris.ly PR is merged, so that link now works.